### DISCUSSION TITLE
How can i divide data in development with production?

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#18327] [Q&A] [answered]
For testing i want to be able to not affect the data that is in production. How can i achieve this?

**Comment by ludralph** [comment]
Hi @gmcamposano 👋 

One way to achieve this is by setting up separate environments for development and production. The development environment would be used for testing and making changes to the application, while the production environment would host the live application and its data.
You can create two different schema.prisma in separate folders and initialise PrismaClient for each schema.prisma that will point to the specific database. Take a look at the workaround in this [issue](https://github.com/prisma/prisma/issues/2443#issuecomment-630679118)

**Reply by nurul3101** [reply]
@gmcamposano You might also want to have a look at Snaplet: https://www.snaplet.dev/
You can get a subset of anonymised data in development from your production data if that's what you meant.

**Comment by gmcamposano** [comment, marked as answer]
@ludralph @nurul3101 I had tried doing this:
```
datasource db {
  provider = "postgresql"
  // url      = env("DATABASE_URL")
  url      = NODE_ENV === "production" ? env("DATABASE_URL") : "postgresql://postgres:postgres@localhost:5432/pos?schema=public"
}
```
Which I thought might work but it didn't.

**Reply by nurul3101** [reply]
You can use multiple env files to switch database URLs for multiple environments.
Here's a [guide](https://www.prisma.io/docs/guides/development-environment/environment-variables/using-multiple-env-files#setup-multiple-env-files) on how to do that.

**Reply by nurul3101** [reply]
@gmcamposano ☝️ Does it help? Please let us know in case you are still facing any issues. If your query is resolved, can you please mark the discussion as answered?

**Reply by gmcamposano** [reply]
Hi @nurul3101 Sorry for late response. I tried this yesterday, but i don't understand how production works. So i run have the /dev database in .env.development, but how will nextjs distinguish when to run in development or production as I can't pass a NODE_ENV variable with different db urls?
So in my case i'll have two, one locally which runs docker and the production which is going to be handled with post scripts ran in with build.

**Reply by nurul3101** [reply]
You would have two separate branches, right?
One branch would be for development and the other branch for production. (main/master branch)
The env file for development can be called `.env.development`, and the production one can be called `.env.production`
Both these env files will have the `DATABASE_URL` variable.

Then, In your package.json file for the development branch you can specify that `.env.development` env file is to be used and in the production branch you can specify that `.env.production` env file should be used using the dotenv package. ([example](https://www.prisma.io/docs/guides/development-environment/environment-variables/using-multiple-env-files#migration-script))

Does it help?

**Reply by gmcamposano** [reply]
That last one made it perfectly clear. Thank you @nurul3101 
This is mentioned in the docs: .env.production is omitted from the above list as it's is not recommended to store your production credentials locally, even if they are git ignored.
This is the part that confuses me... on how the production db is setup if i can't have and .env.production
